File: TestLLM/api_client_examples.js
/**
 * Phi-3 Mini API Client for Node.js/JavaScript
 * How to call your API from JavaScript projects
 */

// For Node.js - install with: npm install node-fetch
// const fetch = require('node-fetch');

// For browsers - use built-in fetch

class Phi3AlertAPI {
    constructor(baseUrl = 'http://localhost:8000') {
        this.baseUrl = baseUrl;
    }

    async healthCheck() {
        try {
            const response = await fetch(`${this.baseUrl}/health`);
            return await response.json();
        } catch (error) {
            return { error: error.message, status: 'unhealthy' };
        }
    }

    async askQuestion(question, temperature = 0.7, maxLength = 512) {
        try {
            const response = await fetch(`${this.baseUrl}/ask`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({
                    question: question,
                    temperature: temperature,
                    max_length: maxLength
                })
            });

            if (!response.ok) {
                throw new Error(`HTTP error! status: ${response.status}`);
            }

            return await response.json();
        } catch (error) {
            return { error: error.message, status: 'failed' };
        }
    }

    async analyzeAlert(alertMessage) {
        const result = await this.askQuestion(`Alert: ${alertMessage}`);
        
        if (result.status === 'success') {
            return result.answer;
        } else {
            return `Error: ${result.error || 'Unknown error'}`;
        }
    }
}

// Usage Examples
async function main() {
    const api = new Phi3AlertAPI();
    
    // 1. Check API health
    console.log('🔍 Checking API health...');
    const health = await api.healthCheck();
    console.log('Status:', health);
    console.log();

    if (health.status !== 'healthy') {
        console.log('❌ API is not healthy. Make sure the server is running!');
        return;
    }

    // 2. Test alert analysis
    const alerts = [
        'High memory usage on web-server-01, 96% utilized',
        'Database query timeout on prod-mysql-01',
        'Disk space critical: 95% full on /var/log',
        'Network latency spike: 500ms average response time'
    ];

    for (let i = 0; i < alerts.length; i++) {
        const alert = alerts[i];
        console.log(`🚨 Alert ${i + 1}: ${alert}`);
        console.log('-'.repeat(50));

        const response = await api.askQuestion(alert, 0.5, 200);
        
        if (response.status === 'success') {
            console.log(`💡 Response: ${response.answer.substring(0, 200)}...`);
            console.log(`⏱️  Time: ${response.response_time}s`);
        } else {
            console.log(`❌ Error: ${response.error}`);
        }
        
        console.log('\n' + '='.repeat(60) + '\n');
    }
}

// For Node.js
if (typeof module !== 'undefined' && module.exports) {
    module.exports = Phi3AlertAPI;
    
    // Run examples if this file is executed directly
    if (require.main === module) {
        main().catch(console.error);
    }
}

// For browsers - attach to window
if (typeof window !== 'undefined') {
    window.Phi3AlertAPI = Phi3AlertAPI;
}

File: TestLLM/api_client_examples.py
#!/usr/bin/env python3
"""
Phi-3 Mini API Client Examples
How to call your API from other Python projects
"""

import requests
import json
from typing import Optional

class Phi3AlertAPI:
    def __init__(self, base_url: str = "http://localhost:8000"):
        self.base_url = base_url
    
    def health_check(self) -> dict:
        """Check if API is healthy"""
        try:
            response = requests.get(f"{self.base_url}/health")
            response.raise_for_status()
            return response.json()
        except requests.exceptions.RequestException as e:
            return {"error": str(e), "status": "unhealthy"}
    
    def ask_question(self, question: str, temperature: float = 0.7, max_length: int = 512) -> dict:
        """Ask a question to the Phi-3 model"""
        try:
            payload = {
                "question": question,
                "temperature": temperature,
                "max_length": max_length
            }
            
            response = requests.post(
                f"{self.base_url}/ask",
                json=payload,
                headers={"Content-Type": "application/json"},
                timeout=30  # 30 second timeout
            )
            response.raise_for_status()
            return response.json()
            
        except requests.exceptions.RequestException as e:
            return {"error": str(e), "status": "failed"}
    
    def analyze_alert(self, alert_message: str) -> str:
        """Convenience method for alert analysis"""
        result = self.ask_question(f"Alert: {alert_message}")
        
        if result.get("status") == "success":
            return result["answer"]
        else:
            return f"Error: {result.get('error', 'Unknown error')}"

# Usage Examples
def main():
    # Create API client
    api = Phi3AlertAPI()
    
    # 1. Check if API is running
    print("🔍 Checking API health...")
    health = api.health_check()
    print(f"Status: {health}")
    print()
    
    if health.get("status") != "healthy":
        print("❌ API is not healthy. Make sure the server is running!")
        return
    
    # 2. Test alert questions
    alerts = [
        "High CPU usage on prod-web-01, 95% for 10 minutes",
        "Database connection timeout on prod-db-01", 
        "SSL certificate expiring in 3 days for api.company.com",
        "Memory usage at 98% on payment-service pod"
    ]
    
    for i, alert in enumerate(alerts, 1):
        print(f"🚨 Alert {i}: {alert}")
        print("-" * 50)
        
        # Get response
        response = api.ask_question(alert, temperature=0.5, max_length=200)
        
        if response.get("status") == "success":
            print(f"💡 Response: {response['answer'][:200]}...")
            print(f"⏱️  Time: {response['response_time']}s")
        else:
            print(f"❌ Error: {response}")
        
        print("\n" + "="*60 + "\n")

if __name__ == "__main__":
    main()
